app/views/directives/edit-environment-from.html

```diff
@@ -0,0 +1,94 @@
+<ng-form name="forms.editEnvironmentFrom" novalidate>
+  <div
+    ng-if="$ctrl.showHeader"
+    class="key-value-editor-entry key-value-editor-entry-header">
+    <div class="form-group key-value-editor-header value-header">
+      <div class="input-group">
+        <span class="help-block">{{$ctrl.selectorPlaceholder}}</span>
+      </div>
+    </div>
+  </div>
+
+  <div ng-model="$ctrl.entries" class="key-value-editor" as-sortable="dragControlListeners">
+    <div
+      class="key-value-editor-entry"
+      ng-class-odd="'odd'"
+      ng-class-even="'even'"
+      ng-repeat="entry in $ctrl.envFromEntries"
+      as-sortable-item>
+
+      <div
+        class="form-group key-value-editor-input"
+        ng-class="{ 'has-error': (forms.editEnvironmentFrom[uniqueForValue(unique, $index)].$invalid && forms.editEnvironmentFrom[uniqueForValue(unique, $index)].$touched) }">
+        <div ng-if="$ctrl.isEnvFromReadonly(entry)" class="faux-input-group">
+          <div class="faux-form-control readonly">
+            Set to values in {{entry.selectedEnvFrom.kind | humanizeKind : true | lowercase}}
+            <span
+              ng-if="!('configmaps' | canI : 'get') || !('secrets' | canI : 'get')">
```

Comment:
This doesn't look right. If the value is a config map, we should be able to link to that even if I can't view secrets.

Reply:
Can adjust the logic, look at separating it out